### PR TITLE
added new chat example

### PR DESCRIPTION
**Pull request by jvsteiner**
Hey, I submitted this a while back, but every time i come back to it and try to update it to work with the current version of stuff, some of the deps change, or it's something else.  The code structure is solid, it basically works, and even if I were to make it work completely, there are no vendored dependencies in this repo, so anyone checking it out, is highly unlikely to be able to run it with no issues anyway.  LMK what you would like, but I'll submit this here in good faith, just to get somewhere....

### CODE REVIEW

File: routed-echo/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Routed Host: echo client/server
+
+This example is intended to follow up the basic host and echo examples by adding use of the ipfs distributed hash table to lookup peers.
+
+Functionally, this example works similarly to the echo example, however setup of the host includes wrapping it with a Kademila hash table, so it can find peers using only their IDs. 
+
+We'll also enable NAT port mapping to illustrate the setup, although it isn't guaranteed to actually be used to make the connections.  Additionally, this example uses the newer `bhost.NewHost` constructor.
```
Comment (tobowers) [inline routed-echo/README.md]: I don't see bhost.NewHost anywhere :)

Comment (jvsteiner) [reply routed-echo/README.md]: hmm, your right.  I'll sort it out, sorry

**Comment by jvsteiner** [comment]
I updated the README to reflect the _actual_ constructor that I meant. There have been a few, and it seems my README documentation had not been updated to refer to the one I am currently actually using.

**Comment by anacrolix** [comment]
LGTM

**Comment by anacrolix** [comment]
@jvsteiner Probably worth using the new DefaultBootstrapPeers list in the dht package if you want to make a PR for that.

**Pull request merged**